### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by colinoflynn**
This is my updates to the documentation to reflect my more major changes. There are a number of TODOs in there still, will continue to update but wanted to have this PR active for easier commenting from everyone.

You can see the mdbook version of these at https://newaetech.github.io/sonata-system/ which is easier to review.

### CODE REVIEW

**Comment by colinoflynn** [comment]
Also - I can squash all these commits into a single one as the commit messages were mostly in-progress work. I didn't want to do that before reviewing though as it may be helpful not having it all squashed to see the specific changes.

**Review by HU90m — DISMISSED** [review]

File: doc/guide/fpga-update.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+> TODO - do we want to recommend a different slot?
+
+## Drag & Drop Programming
+
+> TODO: Where is this bitstream located? Assume it's built by CI, or do we have a release version?
+
+To program the Sonata bitstream:
+
+1. Download `lowrisc_sonata_system_0.bit` from THEBESTURL.COM/lowrisc_sonata_system_0.bit
+2. Select slot 2 using switch SW7 (`Bitstream`)
```
[the remark this answers is not in the document]
Comment (colinoflynn) [reply doc/guide/fpga-update.md:32]: Ah - I was looking at the old revision PCB when I wrote this.

[the remark this answers is not in the document]
Comment (colinoflynn) [reply doc/guide/fpga-update.md]: Fixed now, thanks for catching that!

File: doc/dev/developing.md [hunk 2]
```diff
@@ -46,6 +46,40 @@ In short, these are our general requirements:
 ## Detailed specifications
 
 Please have a look at the following documents for more detailed architecture specifications:
+
 - [Physical board](board.md)
 - [FPGA configuration](fpga.md)
 - [Software](software.md)
```
Comment (marnovandermaas) [inline doc/dev/developing.md:52]: We can remove these bullet points as they are remnants from the old documentation

File: doc/dev/fpga-development.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (marnovandermaas) [inline doc/dev/fpga-development.md:1]:
```suggestion
# FPGA development

## Dependencies
```

File: doc/guide/fpga-update.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+Release Done   0
+Done           0
+ID Error       ID error
+DEC Error      0
+XADC Over temp 0
+STARTUP State  0
+Reserved       0
+BUS Width      x1
+Reserved       8
+```
```
Comment (marnovandermaas) [inline doc/guide/fpga-update.md:138]: This is repeated from the FPGA programming section, maybe just leave it there for now?

**Review by marnovandermaas — COMMENTED** [review]
Just a few comments from my side. Once those are resolved I'm happy to merge this and iterate on it after that.

Also, we need to make sure to squash this before merging.

File: doc/guide/toolchain-setup.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+# Build the toolchain
+cmake ${LLVM_PATH}/llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;lld" -DCMAKE_INSTALL_PREFIX=install -DLLVM_ENABLE_UNWIND_TABLES=NO -DLLVM_TARGETS_TO_BUILD=RISCV -DLLVM_DISTRIBUTION_COMPONENTS="clang;clangd;lld;llvm-objdump;llvm-objcopy" -G Ninja
+export NINJA_STATUS='%p [%f:%s/%t] %o/s, %es'
+ninja install-distribution
+```
+
+> NOTE: Currently the WSL2 build is broken and requires a patch applied as follows:
+> ```
+> wget https://github.com/llvm/llvm-project/commit/4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
+> git am < 4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
```
Comment (HU90m) [inline doc/guide/toolchain-setup.md:70]: nit:
```suggestion
> ```sh
> wget https://github.com/llvm/llvm-project/commit/4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
> git am < 4ad9ec8a328ccb3b836c993bba954366f05b2fd4.patch
> ```
```

**Comment by colinoflynn** [comment]
Thanks all - I'll get these fixed up for tonight my time (aka probably tomorrow for you) & squash it down to a nice single commit once I do. Will flag it back once I think it's ready!

File: doc/guide/fpga-update.md (hunk 4 above)
Comment (colinoflynn) [reply doc/guide/fpga-update.md:138]: I made a short note you can use this feature for debug & linked to the fpga programming section.

File: doc/dev/developing.md (hunk 2 above)
Comment (marnovandermaas) [reply doc/dev/developing.md]: Thanks Colin!

File: doc/guide/fpga-update.md (hunk 4 above)
Comment (marnovandermaas) [reply doc/guide/fpga-update.md:138]: Ok, sounds good.

**Comment by colinoflynn** [comment]
It has been done. https://newaetech.github.io/sonata-system/ was built successfully from our repo, so can use that for checking any changes.

**Review by marnovandermaas — APPROVED** [review]
This looks like a good start to me. Thanks for starting this documentation rewrite. Let's get this into the main repository and iterate from there.

**Pull request merged**